wpilibj/src/main/java/edu/wpi/first/wpilibj/GenericHID.java

```diff
@@ -335,7 +344,10 @@ public BooleanEvent povCenter(EventLoop loop) {
    * @return an event instance that is true when the axis value is less than the provided threshold.
    */
   public BooleanEvent axisLessThan(int axis, double threshold, EventLoop loop) {
-    return new BooleanEvent(loop, () -> getRawAxis(axis) < threshold);
+    var cache = m_axisCache.computeIfAbsent(loop, k -> new HashMap<>());
+    return cache.computeIfAbsent(
+        new Pair<>(axis, -threshold),
```

Comment:
Negating won't be enough to disambiguate `axisLessThan` and `axisGreaterThan`, since the threshold could be negative.